pkgs/development/python-modules/orange3/default.nix

```diff
@@ -0,0 +1,168 @@
+  };
+
+  pythonRelaxDeps = [
+    "scikit-learn"
+    "pyqtgraph" # TODO: remove after bump with a version greater than 0.13.1
+  ];
+
+  postPatch = ''
+    sed -i 's;\(scikit-learn\)[^$]*;\1;g' requirements-core.txt
+    sed -i 's;pyqtgraph[^$]*;;g' requirements-gui.txt
+  '';
+
+  nativeBuildInputs = [
+    copyDesktopItems
+    cython
+    pythonRelaxDepsHook
+    qt5.wrapQtAppsHook
+    sphinx
+  ];
+
+  enableParallelBuilding = true;
+
+  propagatedBuildInputs = [
+    numpy
+    scipy
+    chardet
+    openpyxl
+    opentsne
+    qtconsole
+    bottleneck
+    matplotlib
+    joblib
+    requests
+    keyring
+    scikit-learn
+    pandas
+    pyqtwebengine
+    serverfiles
+    orange-canvas-core
+    python-louvain
+    xlrd
+    xlsxwriter
+    httpx
+    pyqtgraph
+    orange-widget-base
+    keyrings-alt
+    pyyaml
+    baycomp
+  ];
+
+  # FIXME: ImportError: cannot import name '_variable' from partially initialized module 'Orange.data' (most likely due to a circular import) (/build/source/Orange/data/__init__.py)
+  dontCheck = true;
+  pythonImportsCheck = [ "Orange" "Orange.data._variable" ];
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "orange";
+      exec = "orange-canvas";
+      desktopName = "Orange Data Mining";
+      genericName = "Data Mining Suite";
+      comment = "Explore, analyze, and visualize your data";
+      icon = "orange-canvas";
+      mimeTypes = [ "application/x-extension-ows" ];
+      categories = [ "Science" "Education" "ArtificialIntelligence" "DataVisualization" "NumericalAnalysis" "Qt" ];
+      keywords = [ "Machine Learning" "Scientific Visualization" "Statistical Analysis" ];
+    })
+  ];
+
+  postInstall = ''
+    wrapProgram $out/bin/orange-canvas \
+      "${"$"}{qtWrapperArgs[@]}"
+    mkdir -p $out/share/icons/hicolor/{256x256,48x48}/apps
+    cp distribute/icon-256.png $out/share/icons/hicolor/256x256/apps/orange-canvas.png
+    cp distribute/icon-48.png $out/share/icons/hicolor/48x48/apps/orange-canvas.png
+  '';
+
+  passthru = {
+    updateScript = nix-update-script { };
+    tests.unittests = buildPythonPackage {
+      pname = "${pname}-tests";
+      inherit version src;
+      format = "other";
+
+      preCheck = ''
+        export HOME=$(mktemp -d)
+        export QT_PLUGIN_PATH="${qt5.qtbase.bin}/${qt5.qtbase.qtPluginPrefix}"
+        export QT_QPA_PLATFORM_PLUGIN_PATH="${qt5.qtbase.bin}/lib/qt-${qt5.qtbase.version}/plugins";
+        export QT_QPA_PLATFORM=offscreen
+
+        rm Orange -rf
+        cp -r ${orange3}/${python.sitePackages}/Orange .
+        chmod +w -r .
+
+        rm Orange/tests/test_url_reader.py # uses network
+        rm Orange/tests/test_ada_boost.py # broken: The 'base_estimator' parameter of AdaBoostRegressor must be an object implementing 'fit' and 'predict' or a str among {'deprecated'}. Got None instead.
+      '';
+
+      checkPhase = ''
+        runHook preCheck
+        ${python.interpreter} -m unittest -b -v ./Orange/**/test*.py
```

Comment:
Why not the `unittestCheckHook`?

Reply:
unittest discover causes import issues

I am then running the tests based on files

Example of failed case:

```
python3.10-orange3-tests> /build/source/Orange/tests/test_doctest.py:80: UserWarning: Unimportable module: Orange.data.sql.backend.postgres
python3.10-orange3-tests>   warnings.warn('Unimportable module: {}'.format(module))
python3.10-orange3-tests> /build/source/Orange/tests/test_doctest.py:80: UserWarning: Unimportable module: Orange.data.sql.backend.mssql
python3.10-orange3-tests>   warnings.warn('Unimportable module: {}'.format(module))
python3.10-orange3-tests> /build/source/Orange/tests/test_doctest.py:80: UserWarning: Unimportable module: Orange.regression.xgb
python3.10-orange3-tests>   warnings.warn('Unimportable module: {}'.format(module))
python3.10-orange3-tests> /build/source/Orange/tests/test_doctest.py:80: UserWarning: Unimportable module: Orange.regression.catgb
python3.10-orange3-tests>   warnings.warn('Unimportable module: {}'.format(module))
python3.10-orange3-tests> /build/source/Orange/tests/test_doctest.py:80: UserWarning: Unimportable module: Orange.modelling.xgb
python3.10-orange3-tests>   warnings.warn('Unimportable module: {}'.format(module))
python3.10-orange3-tests> /build/source/Orange/tests/test_doctest.py:80: UserWarning: Unimportable module: Orange.modelling.catgb
python3.10-orange3-tests>   warnings.warn('Unimportable module: {}'.format(module))
python3.10-orange3-tests> /build/source/Orange/tests/test_doctest.py:80: UserWarning: Unimportable module: Orange.classification.catgb
python3.10-orange3-tests>   warnings.warn('Unimportable module: {}'.format(module))
python3.10-orange3-tests> /build/source/Orange/tests/test_doctest.py:80: UserWarning: Unimportable module: Orange.classification.xgb
python3.10-orange3-tests>   warnings.warn('Unimportable module: {}'.format(module))
python3.10-orange3-tests> Traceback (most recent call last):
python3.10-orange3-tests>   File "/nix/store/9c03r86hcdn43dm3hsgjirifvyzfkhwh-python3-3.10.12/lib/python3.10/runpy.py", line 196, in _run_module_as_main
python3.10-orange3-tests>     return _run_code(code, main_globals, None,
python3.10-orange3-tests>   File "/nix/store/9c03r86hcdn43dm3hsgjirifvyzfkhwh-python3-3.10.12/lib/python3.10/runpy.py", line 86, in _run_code
python3.10-orange3-tests>     exec(code, run_globals)
python3.10-orange3-tests>   File "/nix/store/9c03r86hcdn43dm3hsgjirifvyzfkhwh-python3-3.10.12/lib/python3.10/unittest/__main__.py", line 18, in <module>
python3.10-orange3-tests>     main(module=None)
python3.10-orange3-tests>   File "/nix/store/9c03r86hcdn43dm3hsgjirifvyzfkhwh-python3-3.10.12/lib/python3.10/unittest/main.py", line 100, in __init__
python3.10-orange3-tests>     self.parseArgs(argv)
python3.10-orange3-tests>   File "/nix/store/9c03r86hcdn43dm3hsgjirifvyzfkhwh-python3-3.10.12/lib/python3.10/unittest/main.py", line 124, in parseArgs
python3.10-orange3-tests>     self._do_discovery(argv[2:])
python3.10-orange3-tests>   File "/nix/store/9c03r86hcdn43dm3hsgjirifvyzfkhwh-python3-3.10.12/lib/python3.10/unittest/main.py", line 244, in _do_discovery
python3.10-orange3-tests>     self.createTests(from_discovery=True, Loader=Loader)
python3.10-orange3-tests>   File "/nix/store/9c03r86hcdn43dm3hsgjirifvyzfkhwh-python3-3.10.12/lib/python3.10/unittest/main.py", line 154, in createTests
python3.10-orange3-tests>     self.test = loader.discover(self.start, self.pattern, self.top)
python3.10-orange3-tests>   File "/nix/store/9c03r86hcdn43dm3hsgjirifvyzfkhwh-python3-3.10.12/lib/python3.10/unittest/loader.py", line 349, in discover
python3.10-orange3-tests>     tests = list(self._find_tests(start_dir, pattern))
python3.10-orange3-tests>   File "/nix/store/9c03r86hcdn43dm3hsgjirifvyzfkhwh-python3-3.10.12/lib/python3.10/unittest/loader.py", line 414, in _find_tests
python3.10-orange3-tests>     yield from self._find_tests(full_path, pattern, namespace)
python3.10-orange3-tests>   File "/nix/store/9c03r86hcdn43dm3hsgjirifvyzfkhwh-python3-3.10.12/lib/python3.10/unittest/loader.py", line 405, in _find_tests
python3.10-orange3-tests>     tests, should_recurse = self._find_test_path(
python3.10-orange3-tests>   File "/nix/store/9c03r86hcdn43dm3hsgjirifvyzfkhwh-python3-3.10.12/lib/python3.10/unittest/loader.py", line 468, in _find_test_path
python3.10-orange3-tests>     name = self._get_name_from_path(full_path)
python3.10-orange3-tests>   File "/nix/store/9c03r86hcdn43dm3hsgjirifvyzfkhwh-python3-3.10.12/lib/python3.10/unittest/loader.py", line 371, in _get_name_from_path
python3.10-orange3-tests>     assert not _relpath.startswith('..'), "Path must be within the project"
python3.10-orange3-tests> AssertionError: Path must be within the project
```

Reply:
But I think because of a suggestion the recursive flag of that chmod is wrong

Served as a double check lol